code/modules/overmap/ships/machines/gas_thruster.dm

```diff
@@ -0,0 +1,73 @@
+/obj/machinery/atmospherics/unary/engine
+	name = "rocket nozzle"
+	desc = "Simple rocket nozzle, expelling gas at hypersonic velocities to propell the ship."
+	icon = 'icons/obj/ship_engine.dmi'
+	icon_state = "nozzle"
+	opacity = 1
+	density = 1
+	atmos_canpass = CANPASS_NEVER
+	connect_types = CONNECT_TYPE_REGULAR|CONNECT_TYPE_FUEL
+	var/engine_extension = /datum/extension/ship_engine/gas
+	construct_state = /decl/machine_construction/default/panel_closed
+	maximum_component_parts = list(/obj/item/stock_parts = 6)//don't want too many, let upgraded component shine
+	uncreated_component_parts = list(/obj/item/stock_parts/power/apc/buildable = 1)
```

Comment:
I think `uncreated_component_parts` is for invisible non-interactive stuff like hidden tesla relays in non-buildable machines. It'll always create these even on construction.
If you meant the machine should spawn with this when mapped, it's set on the circuit, `additional_spawn_components`